File: memory-protocol/src/main/java/io/openapiparser/memory/Memory.java
/*
 * Copyright 2021 https://github.com/openapi-processor/openapi-parser
 * PDX-License-Identifier: Apache-2.0
 */

package io.openapiparser.memory;

import java.util.concurrent.ConcurrentHashMap;

/**
 * in-memory content by path.
 */
public class Memory {
    private static final ConcurrentHashMap<String, String> contents = new ConcurrentHashMap<> ();

    public static String get(String path) {
        if(!contents.containsKey (path))
            throw new RuntimeException ();

        return contents.get (path);
    }

    public static void add(String path, String data) {
        contents.put (path, data);
    }

    public static void clear() {
        contents.clear ();
    }
}

File: openapi-parser-memory/src/main/kotlin/io/openapiparser/memory/Memory.kt
/*
 * Copyright 2021 https://github.com/openapi-processor/openapi-parser
 * PDX-License-Identifier: Apache-2.0
 */

package io.openapiparser.memory

/**
 * in-memory content by path.
 */
class Memory {
    companion object {
        private val contents: HashMap<String, ByteArray> = hashMapOf()

        fun get(path: String): ByteArray {
            return contents[path] ?: throw Exception()
        }

        fun add(path: String, data: String) {
            add(path, data.toByteArray(Charsets.UTF_8))
        }

        fun clear() {
            contents.clear()
        }

        private fun add(path: String, data: ByteArray) {
            contents[path] = data
        }
    }
}
